docs/current_docs/features/visualization.mdx

```diff
@@ -0,0 +1,75 @@
+
+Here's an example of it in action:
+
+![Dagger Trace](/img/current_docs/quickstart/trace.gif)
+
+Dagger is aware of the context it runs on. The Dagger Cloud dashboard classifies Traces depending on whether you triggered a run from your local development host or from your CI environment. You can select the appropriate view from the dashboard sidebar. Here's an example showing Traces generated in a CI environment:
+
+![Traces list](/img/current_docs/features/trace-list.png)
+
+This initial view displays groups of Traces that belong to the same point in Git history. The grouping is determined by the branch/reference combination where Dagger was triggered. This is useful when you need to inspect all the Traces emitted at a specific, pinned point in the repository's version history
+
+Here's a quick summary of what you'll see for each Trace group:
+
+- Repository name
+- Commit metadata, including comment and commit author
+- Indication of the run status: in progress, succeeded, or failed
+- The command used
+- The run start time
+- The run duration
+
+:::important
+All of the information listed above may not be available for a specific group; it differs depending on whether the Trace is generated by a local run or a CI run. When Dagger runs in a CI context like GitHub, GitLab, CircleCI, or Jenkins, additional Trace metadata is displayed based on the Git repository information available.
+:::
+
+You can drill down into the details of a specific group by clicking it. This directs you to a list of Traces, together with detailed status and duration metadata for each. Each Dagger Function call is represented as a separate Trace, as shown below:
+
+![Trace details](/img/current_docs/features/trace-details.png)
+
+You can click any of the Traces in the list to see detailed logs for each Trace. The flame graph at the top of the Trace shows the operations performed by the Dagger Function, with a visual cue of how much time each operation represents of the total duration, as shown below:
+
+![Trace log](/img/current_docs/features/trace-log.png)
+
+:::info
+Dagger uses [OpenTelemetry](https://opentelemetry.io/) as its telemetry protocol. Each Dagger Function call is represented as an [OpenTelemetry trace](https://opentelemetry.io/docs/concepts/signals/traces/). The operations performed by the Dagger Function are represented as [OpenTelemetry spans](https://opentelemetry.io/docs/concepts/signals/traces/#spans). A span can emit several effects, which are the operations generated in the span. For example, the `withDirectory()` span generates all the necessary operations to provide a directory with excluded and included elements. For this, it first loads a directory, and then creates a new one from scratch inside the container, generating two distinct steps (effects).
+:::
+
+If there are any errors, Dagger Cloud automatically brings you to the first error in the list.
```

Comment:
It doesn't really "take you" there, but it surfaces what we think is the main error of the trace, and the main error message.

Reply:
Modified